### PR TITLE
Bluez: improve get services performance

### LINKED ISSUE #927: Linear searching is a performance bottleneck in `BlueZManager.get_services` with many devices
Properties are indexed well for `_wait_condition` but building the services can be expensive when the number of properties stored gets to a large number.  Its noticeable when there ~20 bluetooth devices that have their services resolved in the properties dict

### PR DESCRIPTION
**Pull request by dlech**
Fixes #927.

See commit message for details.

@bdraco can you test and see if this actually makes a measurable difference in performance?

### CODE REVIEW

File: bleak/backends/bluezdbus/manager.py [hunk 1]
```diff
@@ -699,6 +594,22 @@ def _parse_msg(self, message: Message):
...
+                    char_props = cast(GattCharacteristic1, unpacked_props)
+                    self._characteristic_map.setdefault(
+                        char_props["Service"], set()
+                    ).add(obj_path)
+                elif interface == defs.GATT_DESCRIPTOR_INTERFACE:
+                    desc_props = cast(GattDescriptor1, unpacked_props)
+                    self._descriptor_map.setdefault(
+                        desc_props["Characteristic"], set()
+                    ).add(obj_path)
+
```
Comment (bdraco) [inline bleak/backends/bluezdbus/manager.py:612]: Could make the `if` below an `elif`

**Comment by bdraco** [comment]
This will definitely solve the performance issue.

**Comment by bdraco** [comment]
I did some testing with this and it seems the scanner is killed off when ever an active connection is being made.  Not sure if its a regression on develop

**Comment by bdraco** [comment]
I double checking that on a different system, it could be a hardware issue

**Comment by bdraco** [comment]
Switched to a known good system and testing now

**Comment by bdraco** [comment]
Going to take a while to put batteries in enough bluetooth devices to get to the test level I had before

**Comment by bdraco** [comment]
```
2022-08-26 18:45:51.811 DEBUG (MainThread) [bleak_retry_connector] 60-55-F9-3C-34-DA (60:55:F9:3C:34:DA) - /org/bluez/hci1/dev_60_55_F9_3C_34_DA: Connecting (attempt: 1, last rssi: -47)
2022-08-26 18:45:53.498 DEBUG (MainThread) [bleak_retry_connector] 60-55-F9-3C-34-DA (60:55:F9:3C:34:DA) - /org/bluez/hci1/dev_60_55_F9_3C_34_DA: Connected (attempt: 1, last rssi: -47)
```

connecting to a switchbot seems faster.  digging though the callgrind now

**Comment by bdraco** [comment]
Connect time with services cached -- its a layered one top version of https://github.com/hbldh/bleak/pull/923
```
2022-08-26 18:47:08.666 DEBUG (MainThread) [bleak_retry_connector] WoPlug (60:55:F9:3C:34:DA) - /org/bluez/hci1/dev_60_55_F9_3C_34_DA: Connecting (attempt: 1, last rssi: -43)
2022-08-26 18:47:08.860 DEBUG (MainThread) [bleak_retry_connector] Cached services found: <bleak.backends.service.BleakGATTServiceCollection object at 0xffff9b3e9000>
2022-08-26 18:47:08.860 DEBUG (MainThread) [bleak_retry_connector] WoPlug (60:55:F9:3C:34:DA) - /org/bluez/hci1/dev_60_55_F9_3C_34_DA: Connected (attempt: 1, last rssi: -43)
```

**Comment by bdraco** [comment]
get_services

The overhead just disappears with this change

**Review by bdraco — APPROVED** [review]

**Comment by bdraco** [comment]
Been testing with this in production and the performance improvement is noticeable

**Comment by bdraco** [comment]
This has been a really nice change. I added support for some switchbot lights in https://github.com/home-assistant/core/pull/77430 and the performance improvement was quite noticeable on first connect

**Comment by dlech** [comment]
Thanks for testing.

**Pull request merged**